backend/src/openbeheer/catalogi/api/viewsets.py

```diff
@@ -13,3 +19,63 @@
 )
 class ZaaktypenViewSet(ZGWViewSet):
     service_type = APITypes.ztc
+
+
+@extend_schema(
+    summary=_("Get Open Zaak choices"),
+    description=_(
+        "Get the available Open Zaak catalogi instances. The value is the slug of the configured service, "
+        "while the label is the name of the service."
+    ),
+    # responses={
+    #     "200": list[OBOption[str]] # TODO blueprint of msgspec
+    # }
+)
+class ServiceChoicesView(MsgspecAPIView): # TODO rebase on Chris PR
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request: Request) -> Response:
+        services = Service.objects.filter(api_type=APITypes.ztc)
+
+        data = [{"label": service.label, "value": service.slug} for service in services]
+        return Response(data)
+
+
+@extend_schema(
+    summary=_("Get catalogue choices"),
+    description=_(
+        "Retrieve the catalogues available in an Open Zaak instance. "
+        "The value is the URL of the catalogus as returned from Open Zaak, "
+        "while the label is the name of the catalogue (if configured) and otherwise the domain field."
+    ),
+    # responses={
+    #     "200": list[OBOption[str]] # TODO blueprint of msgspec
+    # }
+)
+class CatalogChoicesView(MsgspecAPIView): # TODO rebase on Chris PR
```

Comment:
We have `ZGWViewSet` which hides away many of the implementation details with regards to zgw services. I think we can:

1.  Refactor this to mostly to something like:

```python
@extend_schema(
    tags=["Catalogi"],
    summary=_("Alle catalogi opvragen."),
    description=_("Deze lijst kan gefilterd wordt met query-string parameters."),
)
class CatalogiViewSet(ZGWViewSet):
    service_api_base = "catalogussen"
    service_type = APITypes.ztc
```

2. Transform to `Option`s on the frontend.
3. Possibly update `ZGWViewSet` to deal with merging pages.
4. Possibly consider an approach to or similar to: https://github.com/maykinmedia/open-beheer/pull/32